config/samples/k8sgpt_localai.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: core.k8sgpt.ai/v1alpha1
+kind: K8sGPT
+metadata:
+  name: k8sgpt-sample-localai
+  namespace: k8sgpt-operator-system
+spec:
+  model: ggml-gpt4all-j-v1.3-groovy.bin
+  backend: localai
+  baseUrl: http://local-ai.local-ai.svc.cluster.local:8080/v1
+  noCache: false
+  enableAI: true
+  version: v0.3.0
+  # if not specified, k8s will marshal as a zero-valued SecretRef, which causes errors
+  secret: null
```

Comment:
@TylerGillson , a bit off-topic with the context of the PR but shouldn't we mark the secret as optional in the CRD spec?

_// +optional
specifies that this field is optional, if fields are required by default._

Reply:
@arbreezy I tried with both `// +optional` and `// +nullable`. Maybe I am making a silly mistake, but the CR ends up being created with `secret: {}` in all cases.

Reply:
The behaviour I am seeing is as if `// +kubebuilder:pruning:PreserveUnknownFields` was set, but it isn't. Not sure why.

Reply:
interesting, what if we made it null as its default value so we cover it from the user?
@TylerGillson wdyt?

Reply:
@arbreezy I have experimented with `// +kubebuilder:default={}`, `// +kubebuilder:default:={}`, `// +kubebuilder:default={secretref:{}}`, etc. and none of these approaches have the desired effect.

Reply:
I am trying to reproduce but I don't see any `secret: {}`